libs/workspace/src/builders/barista-build/renderer.ts

```diff
@@ -13,41 +13,49 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { logging } from '@angular-devkit/core';
+
 import axios from 'axios';
-import { grey } from 'chalk';
 import { mkdirSync, promises as fs } from 'fs';
 import { dirname, join } from 'path';
 
-/** Collects a list of routes and get the html code of it */
-export async function renderRoutes(config: {
-  outputPath: string;
-  baseURL: string;
-  routes: string[];
-  logger?: logging.LoggerApi;
-}): Promise<number> {
-  const max = config.routes.length;
+/** generates the html filename out of a route */
+function generateFileName(route: string): string {
+  return join(route, 'index.html');
+}
+
+/** Renders each route that is provided via the process args */
+(async () => {
+  const [outputPath, baseUrl, ...routes] = process.argv.slice(2);
 
-  for (let i = 0; i < max; i++) {
-    const route = config.routes[i];
-    const { data } = await axios.get<string>(route, {
-      baseURL: config.baseURL,
-    });
+  for (const route of routes) {
     const fileName = generateFileName(route);
-    const filePath = join(config.outputPath, fileName);
+    const filePath = join(outputPath, fileName);
+
+    try {
+      const { data } = await axios.get<string>(route, {
+        baseURL: baseUrl,
+      });
 
-    mkdirSync(dirname(filePath), { recursive: true });
-    await fs.writeFile(filePath, data, 'utf-8');
+      mkdirSync(dirname(filePath), { recursive: true });
+      await fs.writeFile(filePath, data, 'utf-8');
 
-    if (config.logger) {
-      config.logger.info(`CREATED: ${grey(filePath)}`);
+      if (process.send) {
+        process.send({
+          success: true,
+          filePath,
+          size: Buffer.byteLength(data).toFixed(0),
+        });
+      }
+    } catch (error) {
+      if (process.send) {
+        process.send({
+          success: false,
+          message: error.message,
+          filePath,
+        });
+      }
     }
   }
-
-  return max;
-}
-
-/** generates the html filename out of a route */
-function generateFileName(route: string): string {
-  return join(route, 'index.html');
-}
+})()
+  .then()
+  .catch();
```

Comment:
I don't think you need the empty then here
```suggestion
  .catch();
```